pkg/client/restclient/config.go

```diff
@@ -283,11 +302,16 @@ func InClusterConfig() (*Config, error) {
 // Note: the Insecure flag is ignored when testing for this value, so MITM attacks are
 // still possible.
 func IsConfigTransportTLS(config Config) bool {
-	baseURL, _, err := defaultServerUrlFor(&config)
-	if err != nil {
+	hosts, _, err := defaultServerUrlFor(&config)
```

Comment:
I think `defaultServerUrlFor` should return `err != nil` if hosts would be empty.